MyCryptoHQ/MyCrypto
# Enhancement/new metamask stuff

Closes CORE-51.

### Description

Implements the new MetaMask changes as prescribed at https://medium.com/metamask/https-medium-com-metamask-breaking-change-injecting-web3-7722797916a8.

### Changes

When accessing the MetaMask option on MetaMask v4.12.0 or greater, a prompt will occur with "Accept/Reject". If the user clicks "Reject", or the back button, they are sent to the "Unlock MetaMask" screen with an error informing them they should approve. The same dialog occurs if the click the button and reject again. On approval, they are allowed in, and are automatically allowed in each subsequent time unless permission is revoked.

### Steps to Test

1. Open the home screen and have MetaMask 4.12.0 or greater installed.
2. Click the "MetaMask" wallet unlock option.
3. When the dialog pops up, press "Reject".
4. You should be shown the "Unlock MetaMask" screen with an error informing the need to accept.
5. Press "Unlock MetaMask". When the dialog pops up, again press "Reject".
6. The same error message in 4. should be shown.
7. Press "Unlock MetaMask". When the dialog pops up, press the back button at the top left.
8. The same error message in 4. should be shown.
9. Press "Unlock MetaMask". When the dialog pops up, press the "Approve" button.
10. You should be shown the Send Transaction screen.
11. Reload the page.
12. Click the "MetaMask" wallet option.
13. You should be shown the Send Transaction screen.
